libcudacxx/include/cuda/std/detail/libcxx/include/__string

```diff
@@ -254,19 +254,30 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT char_traits<char>
 
   _LIBCUDACXX_HIDE_FROM_ABI static _CCCL_CONSTEXPR_CXX14 int
   compare(const char_type* __s1, const char_type* __s2, size_t __n) noexcept;
-  _LIBCUDACXX_HIDE_FROM_ABI static size_t _CCCL_CONSTEXPR_CXX14 length(const char_type* __s) noexcept
+
+  // Define a fallback _CCCL_BUILTIN_STRLEN function using parens to prevent
+  // macro expansion.
+  _LIBCUDACXX_HIDE_FROM_ABI static _CCCL_CONSTEXPR_CXX14 size_t(_CCCL_BUILTIN_STRLEN)(const char_type* __s) noexcept
```

Comment:
I am strongly against naming this the same. We can name this `__strlen_fallback` but that is a bit much